app/javascript/article-form/elements/imageManagement.jsx

```diff
@@ -17,22 +18,29 @@ export default class ImageManagement extends Component {
     e.preventDefault();
 
     this.clearUploadError();
+    const validFileInputs = validateFileInputs();
 
-    const payload = { image: e.target.files, wrap_cloudinary: true };
-    const { onMainImageUrlChange } = this.props;
+    if (validFileInputs) {
+      const payload = { image: e.target.files, wrap_cloudinary: true };
+      const { onMainImageUrlChange } = this.props;
 
-    generateMainImage(payload, onMainImageUrlChange, this.onUploadError);
+      generateMainImage(payload, onMainImageUrlChange, this.onUploadError);
+    }
```

Comment:
I recognize this probably isn't the _best_ way to handle this situation. It's a good example of me trying to strike a good balance between hack and re-usability. I'm open to suggestions on how to improve this, keeping in mind we'll likely change it entirely in the future.